components/transport-netty-axis2-bridge/src/main/java/org/wso2/transports/http/bridge/sender/AxisToClientConnectorBridge.java

```diff
@@ -161,34 +161,38 @@ private void sendForward(MessageContext msgCtx, HttpCarbonMessage httpCarbonMess
 
         // serialize
         if (Boolean.TRUE.equals(msgCtx.getProperty(BridgeConstants.MESSAGE_BUILDER_INVOKED))) {
-            final HttpMessageDataStreamer outboundMsgDataStreamer = getHttpMessageDataStreamer(httpCarbonMessage);
-            final OutputStream outputStream = outboundMsgDataStreamer.getOutputStream();
-            OMElement omElement = msgCtx.getEnvelope().getBody().getFirstElement();
-            writeToStream(outputStream, omElement.toString());
+            formatAndWrite(msgCtx, httpCarbonMessage);
         }
-
     }
 
-    private void writeToStream(OutputStream outputStream, String omElementString) {
+    private void formatAndWrite(MessageContext msgCtx, HttpCarbonMessage httpCarbonMessage) {
```

Comment:
Shall we rename this to writeWithFormatter or something similar, having and gives the idea that this method is doing multiple stuff.

Reply:
done